scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java

```diff
@@ -0,0 +1,119 @@
+    private final TaskContextVariableExtractor taskContextVariableExtractor = new TaskContextVariableExtractor();
+    private final ForkEnvironmentScriptExecutor forkEnvironmentScriptExecutor = new ForkEnvironmentScriptExecutor();
+
+    public OSProcessBuilder createForkedProcessBuilder(TaskContext context, File serializedContext,
+            PrintStream outputSink, PrintStream errorSink, File workingDir) throws Exception {
+        OSProcessBuilder processBuilder;
+
+        String nativeScriptPath = context.getSchedulerHome();
+        ScriptResult forkEnvironmentScriptResult = null;
+
+        if (context.isRunAsUser()) {
+            shareWorkingDirWithRunAsMeUser(workingDir);
+            processBuilder = ForkerUtils.getOSProcessBuilderFactory(nativeScriptPath).getBuilder(
+                    ForkerUtils.checkConfigAndGetUser(context.getDecrypter()));
+        } else {
+            processBuilder = ForkerUtils.getOSProcessBuilderFactory(nativeScriptPath).getBuilder();
+        }
+
+        ForkEnvironment forkEnvironment = context.getInitializer().getForkEnvironment();
+
+        if (forkEnvironment != null) {
+
+            if (forkEnvironment.getEnvScript() != null) {
+                forkEnvironmentScriptResult = forkEnvironmentScriptExecutor
+                        .executeForkEnvironmentScript(context, outputSink, errorSink);
+            }
+
+            try {
+                processBuilder.environment().putAll(
+                        // replace variables in defined system environment values
+                        // by existing environment variables, variables and credentials
+                        taskContextVariableExtractor
+                                .extractVariablesThirdPartyCredentialsAndSystemEnvironmentVariables(context));
+            } catch (IllegalArgumentException processEnvironmentReadOnly) {
+                throw new IllegalStateException(
+                        "Cannot use runAsMe mode and set system environment properties",
+                        processEnvironmentReadOnly);
+            }
+        }
+
+        processBuilder
+                .command()
+                .addAll(forkedJvmTaskExecutionCommandCreator
+                        .createForkedJvmTaskExecutionCommand(context,
+                                forkEnvironmentScriptResult, serializedContext.getAbsolutePath()
+                        ));
+
+        processBuilder = processBuilder.directory(workingDir);
+        return processBuilder;
+    }
+
+
+    private void shareWorkingDirWithRunAsMeUser(File workingDir) throws IOException {
+        try {
+            Files.setPosixFilePermissions(workingDir.toPath(), SHARED_FOLDER_PERMISSIONS);
+        } catch (IOException e) {
+            throw new IOException("Working directory will not be writable by runAsMe user", e);
+        } catch (UnsupportedOperationException ignored) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)